blocks/software/advanced-controls/controllers/trapezoidal-profiles.yml

```diff
@@ -0,0 +1,65 @@
+block0: |
+  // Creates a new set of trapezoidal motion profile constraints
+  // Max velocity of 10 meters per second
+  // Max acceleration of 20 meters per second squared
+  TrapezoidProfile.Constraints(10.0, 20.0)
+
+block1: |
+  // Creates a new state with a position of 5 meters
+  // and a velocity of 0 meters per second
+  TrapezoidProfile.State(5.0, 0.0)
+
+block2: |
+  // Creates a new TrapezoidProfile
+  // Profile will have a max vel of 5 meters per second
+  // Profile will have a max acceleration of 10 meters per second squared
+  val profile = TrapezoidProfile(TrapezoidProfile.Constraints(5.0, 10.0))
+
+block3: |
+  // Profile will start stationary at zero position
+  // Profile will end stationary at 5 meters
+  // Returns the motion profile state after 5 seconds of motion
+  profile.calculate(5.0, TrapezoidProfile.State(0.0, 0.0), TrapezoidProfile.State(5.0, 0.0))
+
+block4: |
+  val setpoint = profile.calculate(elapsedTime, initialState, goalState)
+  controller.calculate(encoder.getDistance(), setpoint.position)
+
+block5: |
+  private const val DT = 0.02
```

Comment:
I'm wondering if we should put this in a companion object within the robot class. It doesn't make a difference, but I wonder if it's a 'best practice'

Reply:
With regards to this, i think it's more idiomatic to not put it in a companion object(in java the only reason to make it static is to declare it as a "global variable", in kotlin you already have it).